### PR TITLE
- closing messenger instance in the main function

### PR DESCRIPTION
**Pull request by iulianpascalau**
- closing messenger instance in the main function
- added placeholder to insert test code
- added a new integration test for intercepted meta headers

### CODE REVIEW

File: api/node/routes.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ func Routes(router *gin.RouterGroup) {
 	router.GET("/heartbeatstatus", HeartbeatStatus)
 	router.GET("/statistics", Statistics)
 	router.GET("/status", StatusMetrics)
+	// testing section
```
Comment (bogdan-rosianu) [inline api/node/routes.go]: might change this comment as if you leave it like this, the next person will think this is a testing section, but it's the real one instead. Maybe put something like `// placeholder for custom routes`

File: integrationTests/multiShard/block/interceptedBlocks_test.go [hunk 2]
```diff
@@ -176,6 +176,86 @@ func TestMetaHeadersAreRequstedOnlyFromMetachain(t *testing.T) {
...
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	for _, n := range nodes {
+		_ = n.Messenger.Bootstrap()
+	}
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(time.Second * 5)
```
Comment (bogdan-rosianu) [inline integrationTests/multiShard/block/interceptedBlocks_test.go]: might export a constant `const p2pBootstrapDelay = 5 * time.Second`

File: integrationTests/multiShard/block/interceptedBlocks_test.go [hunk 3]
```diff
@@ -195,3 +275,15 @@ func requestAndRetrieveMetaHeader(
...
 	return retrievedObject.(*block.MetaBlock)
 }
+
+func requestAndRetrieveMetaHeadersByNonce(
+	node *integrationTests.TestProcessorNode,
+	nonce uint64,
+) []data.HeaderHandler {
+
+	node.RequestHandler.RequestMetaHeaderByNonce(nonce)
+	time.Sleep(time.Second * 2)
```
Comment (bogdan-rosianu) [inline integrationTests/multiShard/block/interceptedBlocks_test.go]: might export a constant `const delayAfterRequestingHeader = 2 * time.Second`

File: api/node/routes.go (hunk 1 above)
Comment (iulianpascalau) [reply api/node/routes.go]: done

File: integrationTests/multiShard/block/interceptedBlocks_test.go (hunk 2 above)
Comment (iulianpascalau) [reply integrationTests/multiShard/block/interceptedBlocks_test.go]: done

File: integrationTests/multiShard/block/interceptedBlocks_test.go (hunk 3 above)
Comment (iulianpascalau) [reply integrationTests/multiShard/block/interceptedBlocks_test.go]: done

**Pull request merged**